posthog/hogql/test/test_printer.py

```diff
@@ -620,3 +620,13 @@ def test_concat_pipes(self):
             self._expr("'a' || 'b' || 3 || timestamp"),
             f"concat(%(hogql_val_0)s, %(hogql_val_1)s, toString(3), ifNull(toString(toTimeZone(events.timestamp, %(hogql_val_2)s)), ''))",
         )
+
+    def test_tumble_datetime(self):
+        self.assertEqual(
+            self._expr("tumble(toDateTime('2023-06-12'), toIntervalDay('1'))"),
+            f"tumble(assumeNotNull(toDateTime(parseDateTime64BestEffortOrNull(%(hogql_val_0)s, 6, %(hogql_val_1)s))), toIntervalDay(%(hogql_val_2)s))",
+        )
+        self.assertEqual(
+            self._expr("tumble(now(), toIntervalDay('1'))"),
+            f"tumble(toDateTime(now64(6, %(hogql_val_0)s)), toIntervalDay(%(hogql_val_1)s))",
+        )
```

Comment:
Can you add a few more of the cases from here as well?

```sql
SELECT 
  tumble(now(), toIntervalDay('1')),
  tumbleStart(now(), toIntervalDay('1')),
  tumbleEnd(now(), toIntervalDay('1')),
  tumble(toDateTime('2019-01-01 00:00:00'), toIntervalDay('2')),
  tumble(timestamp, toIntervalDay('2')),
  tumble(NOW(), toIntervalDay('1')),
  tumble(parseDateTime('2021-01-04+23:00:00', '%Y-%m-%d+%H:%i:%s'), toIntervalDay('1')),
  tumble(parseDateTimeBestEffort('23/10/2020 12:12:57'), toIntervalDay('1'))
FROM events
```

Also for the `hop*` functions.